packages/data-point/lib/resolve/reducer-function.js

```diff
@@ -1,95 +1,34 @@
 'use strict'
 
-const _ = require('lodash')
 const Promise = require('bluebird')
 const utils = require('../utils')
 
-/**
- * applies reducer parameters and passes them to reducer function,
- * it is expected that the filterFunction will return a function
- * in return with the form of (acc, done) - node style callback
- *
- * @param {any} filterFunction
- * @param {any} reducerParameters
- * @returns
- */
-function getReducerCallback (filterFunction, reducerParameters) {
-  const parameterValues = _.map(reducerParameters, 'value')
-  return filterFunction.apply(null, parameterValues)
-}
-
-module.exports.getReducerCallback = getReducerCallback
-
-/**
- * get callback function from its ObjectStoreManager
- *
- * @param {any} filterStore
- * @param {any} reducerFunction
- * @returns
- */
-function getCallbackFunction (filterStore, reducerFunction) {
-  const filterFunction = filterStore.get(reducerFunction.name)
-  const reducerParameters = reducerFunction.parameters
-
-  return getReducerCallback(filterFunction, reducerParameters)
-}
-
-module.exports.getCallbackFunction = getCallbackFunction
-
 /**
  * Resolve a ReducerFunction
- *
- * @param {ObjectStoreManager} filterStore
  * @param {Accumulator} accumulator
- * @param {ReducerFunction} reducerFunction
+ * @param {ReducerFunction} reducer
  * @returns {Promise<Accumulator>}
  */
-function resolve (filterStore, accumulator, reducerFunction) {
-  const callbackFunction = reducerFunction.isFunction
-    ? reducerFunction.body
-    : getCallbackFunction(filterStore, reducerFunction)
-
-  if (typeof callbackFunction !== 'function') {
-    const e = new Error(
-      `ReducerFunction '${reducerFunction.name}' should return a function`
-    )
-    e.name = 'InvalidType'
-    return Promise.reject(e)
-  }
-
-  if (callbackFunction.length > 2) {
-    const e = new Error(
-      `ReducerFunction '${
-        reducerFunction.name
-      }' must have at most an arity of 2`
-    )
-    e.name = 'InvalidArity'
-    return Promise.reject(e)
-  }
-
-  const currentAccumulator = utils.set(
-    accumulator,
-    'currentFilter',
-    reducerFunction
-  )
+function resolve (accumulator, reducer) {
+  const callbackFunction = reducer.body
 
   if (callbackFunction.length === 2) {
```

Comment:
not for now, but on another iteration, we should cache the type of callback so we don't have to read the length, or even maybe already have a function with it ready to be resolved.... yeah this should be when we 'compile' it...